feat: Update subseries presentation

kesara opened the pull request:
Fixes #1100

With https://github.com/ietf-tools/xml2rfc/pull/1102/commits/205c09c84fb4f204d727ae64c56dab70d61fe768, `xml2rfc` tries to add the correct target for subseries when `referencegroup`'s `target` attribute is missing.

Updated examples:
| Format | Example |
| --- | --- |
| XML | [draft-foobar-00.xml](https://t4.fq.nz/subseries-reference-targets/draft-foobar-00.xml) |
| Text | [draft-foobar-00.txt](https://t4.fq.nz/subseries-reference-targets/draft-foobar-00.txt) |
| HTML | [draft-foobar-00.html](https://t4.fq.nz/subseries-reference-targets/draft-foobar-00.html) |
| PDF | [draft-foobar-00.pdf](https://t4.fq.nz/subseries-reference-targets/draft-foobar-00.pdf) |

ajeanmahoney commented:
This fix's [HTML output](https://t4.fq.nz/subseries-reference/draft-foobar-00.html) matches the current HTML output.  However, the [text output](https://t4.fq.nz/subseries-reference/draft-foobar-00.txt) has line breaks in the URLs, whereas the current layout for text output starts the URL on a new line to prevent some of these breaks.

cabo commented:
(off-topic comment: This appears to be one more case where <nobr functionality would help, this time in the bowels of xml2rfc.)

kesara commented:
> This fix's [HTML output](https://t4.fq.nz/subseries-reference/draft-foobar-00.html) matches the current HTML output. However, the [text output](https://t4.fq.nz/subseries-reference/draft-foobar-00.txt) has line breaks in the URLs, whereas the current layout for text output starts the URL on a new line to prevent some of these breaks.

@ajeanmahoney, I've made appropriate changes. Can you recheck the example outputs? Thanks.

rjsparks commented:
I've approved, but wait for @ajeanmahoney review as well.

ajeanmahoney reviewed (changes requested):
The text output is missing a period between the URL and the "At the time of writing" sentence (the period is there in the HTML and PDF outputs). 

The extrapolated URL (when the target is not given) for a STD, BCP, or FYI should be of the form `<https://www.rfc-editor.org/info/...>`. Currently it shows, for example, `<https://www.rfc-editor.org/std/std1>`

For the non-subseries referencegroup output, I'd like to see one that contains multiple references, both with and without a target, thanks!

kesara commented:
> The text output is missing a period between the URL and the "At the time of writing" sentence (the period is there in the HTML and PDF outputs).
> 

I've fixed this.

> The extrapolated URL (when the target is not given) for a STD, BCP, or FYI should be of the form `<https://www.rfc-editor.org/info/...>`. Currently it shows, for example, `<https://www.rfc-editor.org/std/std1>`
> 

I've fixed this too.

> For the non-subseries referencegroup output, I'd like to see one that contains multiple references, both with and without a target, thanks!

See updated files:
| Format | Example |
| --- | --- |
| XML | [draft-foobar-00.xml](https://t4.fq.nz/subseries-reference-targets/draft-foobar-00.xml) |
| Text | [draft-foobar-00.txt](https://t4.fq.nz/subseries-reference-targets/draft-foobar-00.txt) |
| HTML | [draft-foobar-00.html](https://t4.fq.nz/subseries-reference-targets/draft-foobar-00.html) |
| PDF | [draft-foobar-00.pdf](https://t4.fq.nz/subseries-reference-targets/draft-foobar-00.pdf) |

cabo commented:
Can you do an example that has a referencegroup mixed from references to different subseries?

kesara commented:
> Can you do an example that has a referencegroup mixed from references to different subseries?

@cabo, RPC confirmed that RFCs are not shared between different subseries.
But this is an issue if this happens in the future because the `xml2rfc` logic looks at the first reference within `referencegroup` to identify the subseries.
This is because `referencegroup` element doesn't carry information about the subseries.

cabo commented:
I'm fine with this behavior (first reference determines processing) -- I just wanted to make sure that we have considered this (non-)use case.

kesara commented:
@ajeanmahoney Any issues with the output changes in https://github.com/ietf-tools/xml2rfc/pull/1102#issuecomment-1951542113?

ajeanmahoney reviewed (approved):
I've reviewed the test outputs, and it looks good.
